### ISSUE TITLE
Janrain SDK Causing App to Crash on Launch

### ISSUE THREAD

**Issue by Cordavi** [janrain/jump.ios#56]
The `JRCapture.setCaptureConfig(_:)` function which is called on startup is causing iOS's watchdog to terminate on startup due to a network request blocking the main thread on launch. 

`JRConnectionManager`'s 
```objective-c
(bool)createConnectionFromRequest:(NSURLRequest *)request forDelegate:(id <JRConnectionManagerDelegate>)delegate returnFullResponse:(BOOL)returnFullResponse withTag:(id)userData
``` 
function's semaphore will wait indefenitiely if the request encounters an error like a network time out ( `dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER)`)

There has been a recent increase in this blocking request causing iOS to terminate the app using the SDK on launch.

**Comment by Cordavi** [comment]
Here are some crash reports:
[The Atlantic Crashes.zip](https://github.com/janrain/jump.ios/files/2620851/The.Atlantic.Crashes.zip)

**Comment by bcapps** [comment]
Please also note that blocking the calling thread until a network request completes is heavily discouraged by Apple, and doing so from a call to `setCaptureConfig` is very concerning, as you recommend calling it from `didfinishLaunching` in your [documentation here](https://github.com/janrain/jump.ios/blob/master/Docs/JUMP%20Integration%20Guide.md). This means that startup of any app that integrates this SDK will be delayed until a synchronous network request completes, and if it does not complete quickly for any reason, the app will crash on launch due to the watchdog terminating the app for taking too long to start up.

**Comment by pbjanrain** [comment]
Thank you for the report.  An internal ticket has been created (MOB-554).  Please feel free to reference that ticket number in any communications with support or your account manager regarding this issue.

**Comment by bcapps** [comment]
Thanks @pbjanrain!

**Comment by pbjanrain** [comment]
Should be resolved in 5.1.2 release
